ktlintApplyToIdea fails if android = true

In Version 9.1.1, when `android` is set to `true` then the Gradle task `ktlintApplyToIdea` fails with the following log:

```
> Task :ktlintApplyToIdea FAILED
Exception in thread "main" picocli.CommandLine$UnmatchedArgumentException: Unknown option: --android
	at picocli.CommandLine$Interpreter.parse(CommandLine.java:7283)
	at picocli.CommandLine$Interpreter.processArguments(CommandLine.java:7357)
	at picocli.CommandLine$Interpreter.parse(CommandLine.java:7257)
	at picocli.CommandLine$Interpreter.parse(CommandLine.java:7147)
	at picocli.CommandLine.parseArgs(CommandLine.java:836)
	at com.pinterest.ktlint.Main.main(Main.kt:54)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':ktlintApplyToIdea'.
> Process 'command '/Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
```

```
ktlint {
        android = true
}
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,7 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/).
   - Update Gradle to `6.0.1` version
   - Update Kotlin to `1.3.60` version
 ### Fixed
-  - ?
+  - Fix `ktlintApplyToIdea` task fails when `android = true` is set [#311](https://github.com/JLLeitschuh/ktlint-gradle/issues/311)
 ### Removed
   - ?
 

--- a/plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintApplyToIdeaTask.kt
+++ b/plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintApplyToIdeaTask.kt
@@ -1,6 +1,7 @@
 package org.jlleitschuh.gradle.ktlint
 
 import javax.inject.Inject
+import net.swiftzer.semver.SemVer
 import org.gradle.api.DefaultTask
 import org.gradle.api.file.ConfigurableFileCollection
 import org.gradle.api.model.ObjectFactory
@@ -31,16 +32,36 @@ open class KtlintApplyToIdeaTask @Inject constructor(
         project.javaexec {
             it.classpath = classpath
             it.main = resolveMainClassName(ktlintVersion.get())
-            if (globally.get()) {
-                it.args("--apply-to-idea")
-            } else {
-                it.args("--apply-to-idea-project")
-            }
-            // -y here to auto-overwrite existing IDEA code style
-            it.args("-y")
+
+            // Global flags
             if (android.get()) {
                 it.args("--android")
             }
+
+            // Subcommand
+            if (globally.get()) {
+                it.args(getApplyToIdeaCommand())
+            } else {
+                it.args(getApplyToProjectCommand())
+            }
+
+            // Subcommand parameters
+            // -y here to auto-overwrite existing IDEA code style
+            it.args("-y")
         }
     }
+
+    private fun getApplyToIdeaCommand() =
+        if (SemVer.parse(ktlintVersion.get()) >= SemVer(0, 35, 0)) {
+            "applyToIDEA"
+        } else {
+            "--apply-to-idea"
+        }
+
+    private fun getApplyToProjectCommand() =
+        if (SemVer.parse(ktlintVersion.get()) >= SemVer(0, 35, 0)) {
+            "applyToIDEAProject"
+        } else {
+            "--apply-to-idea-project"
+        }
 }

